resources.policy_set_definitions.tf

```
@@ -14,14 +14,35 @@ resource "azurerm_policy_set_definition" "enterprise_scale" {
         policyDefinitionId          = item.policyDefinitionId
         parameters                  = try(jsonencode(item.parameters), null)
         policyDefinitionReferenceId = try(item.policyDefinitionReferenceId, null)
+        groupNames                  = try(item.groupNames, null)
       }
     ]
     content {
       policy_definition_id = policy_definition_reference.value["policyDefinitionId"]
       parameter_values     = policy_definition_reference.value["parameters"]
       reference_id         = policy_definition_reference.value["policyDefinitionReferenceId"]
+      policy_group_names   = policy_definition_reference.value["groupNames"]
     }
   }
+dynamic "policy_definition_group" {
+  for_each = [for item in coalesce(each.value.template.properties.policyDefinitionGroups, []) :
+    {
+      name                 = item.name
+      displayName          = try(item.displayName, null)
+      description          = try(item.description, null)
+      category             = try(item.category, null)
+      additionalMetadataId = try(item.additionalMetadataId, null)
+    }
```

Do we need to check that `item.name` is not null or empty?
```suggestion
    } if item.name != null or item.name != ""
```

---

Added the statement if item.name != null && item.name != "" at line 35 and tested